### PR TITLE
Move TPraos protocol into its own package.

### PR DESCRIPTION
**Pull request by nc6**
Intention will be to implement the new Praos protocol also in this package.

This PR simply moves things around. I didn't move the PBFT protocol into this package, since it seemed reasonable to just isolate it in Byron.

### CODE REVIEW

File: ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    cardano-ledger-shelley,
+    cardano-protocol-tpraos,
+    cardano-slotting,
+    cborg,
+    containers,
+    mtl,
+    nothunks,
+    ouroboros-consensus,
+    serialise,
+    text
```
Comment (nfrisby) [inline ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal:34]: I suspect we also want to copy over the rest of the litany, especially `ghc-options`.

File: ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Ledger/HotKey.hs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 -- | Hot key
 --
 -- Intended for qualified import
-module Ouroboros.Consensus.Shelley.Protocol.HotKey (
+module Ouroboros.Consensus.Protocol.Ledger.HotKey (
```
Comment (nfrisby) [inline ouroboros-consensus-protocol/src/Ouroboros/Consensus/Protocol/Ledger/HotKey.hs:11]: Thinking out loud: is the hotkey more a `Ledger` concern or more a `BlockForging` concern?

Actually, I'm less sure how to distinguish which parts of this are unique to `BlockForging` than I had thought I was. EG only a block-producing node would ever use `evolve`, `sign`, and `mkHotKey`, if I understand correctly.

I suppose it's natural to leave this module unbroken, and so `Ledger` really is the most appropriate point in the conceptual hierarchy for this bag of definitions.

**Review by nfrisby — COMMENTED** [review]
Approved. Thanks!

The only key request is to copy over the `ghc-options` block.

**Review by nfrisby — APPROVED** [review]
Approved! Sorry -- I meant to Approve instead of just Comment last time.

**Comment by nc6** [comment]
bors merge

**Comment by nc6** [comment]
bors retry

**Pull request merged**